pkg/daemon/daemon.go

```diff
@@ -413,6 +413,8 @@ func (dn *Daemon) nodeStateSyncHandler(generation int64) error {
 				syncStatus:    "Succeeded",
 				lastSyncError: "",
 			}
+			// wait for writer to refresh the status
+			<-dn.syncCh
```

Comment:
IIUC as SyncCh is initialized with a zero capacity. then [1] will block until someone is trying to read from the channel.

nice catch here @rollandf 

[1] https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/815fd134ba8000756791051fca60179ec66ddb46/pkg/daemon/writer.go#L98